src/ncbitaxon.py

```diff
@@ -141,7 +152,9 @@ def convert_node(node, label, merged, synonyms, citations):
     rank = node["rank"]
     if rank and rank != "" and rank != "no rank":
         if rank not in ranks:
-            print(f"WARN Unrecognized rank '{rank}'")
+            if rank not in UNRECOGNIZED_RANKS:
```

Comment:
It's missing `keys()` function in the `UNRECOGNIZED_RANKS`. Otherwise it lists every entry which is what you wanted to avoid here, right?

Reply:
Using `.keys()` for checking membership is a common anti-pattern. Any object `y` that implements `__in__` can use the `x in y` or `x not y` construction.

`UNRECOGNIZED_RANKS` is a `collections.Counter`, which indeed implements `__in__` as all mappings do.

Reply:
It didn't work as expected here.

Reply:
I removed some of the ranks you added when I tested the code.

Reply:
You didn't need to create PR #91. That's fine.

Reply:
Let's continue discussion about logging in #91 so we can finish this PR, which is now exclusively about material changes to the rank list